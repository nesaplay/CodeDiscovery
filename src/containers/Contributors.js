import React, { Component } from 'react'

export default class Contributors extends Component {
	render() {
		return <div>Contributors container</div>
	}
}
