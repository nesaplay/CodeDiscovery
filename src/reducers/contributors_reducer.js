import {
	FETCH_CONTRIBUTORS,
	FETCH_CONTRIBUTORS_SUCCESS,
    FETCH_CONTRIBUTORS_ERROR,
    CONTRIBUTORS_PAGE_UP,
    RESET_CONTRIBUTORS
} from '../actions/types'

const initState = {
	data: [],
	page: 1,
	isLoading: false,
	isLoaded: false,
	error: null
}

export default function(state = initState, action) {
	switch (action.type) {
		case FETCH_CONTRIBUTORS:
			return {
				...state,
                isLoading: true,
                isLoaded: false,
				error: null
			}
		case FETCH_CONTRIBUTORS_SUCCESS:
			return {
				...state,
				data: [...state.data, ...action.payload],
				isLoading: false,
				isLoaded: true
			}
		case FETCH_CONTRIBUTORS_ERROR:
			return {
				...state,
				isLoading: false,
				isLoaded: true,
				error: action.error
			}
		case CONTRIBUTORS_PAGE_UP:
			return {
				...state,
				page: state.page + 1
			}
		case RESET_CONTRIBUTORS:
			return initState
		default:
			return state
	}
}
