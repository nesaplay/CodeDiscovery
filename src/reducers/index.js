import { combineReducers } from 'redux'
import search from './search_reducer'
import repositories from './repositories_reducer'

const rootReducer = combineReducers({
    search,
    repositories
})

export default rootReducer
