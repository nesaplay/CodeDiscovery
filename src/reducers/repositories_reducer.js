import {
	FETCH_REPOSITORIES,
	FETCH_REPOSITORIES_SUCCESS,
	FETCH_REPOSITORIES_ERROR,
    RESET_REPOSITORIES
} from '../actions/types'

const initState = {
	data: [],
	isLoading: false,
	isLoaded: false,
	error: null
}

export default function(state = initState, action) {
	switch (action.type) {
		case FETCH_REPOSITORIES:
			return {
				...state,
                isLoading: true,
                isLoaded: false,
				error: null
			}
		case FETCH_REPOSITORIES_SUCCESS:
			return {
				...state,
				data: action.payload,
				isLoading: false,
				isLoaded: true
			}
		case FETCH_REPOSITORIES_ERROR:
			return {
				...state,
				isLoading: false,
				isLoaded: false,
				error: action.error.message
			}
		case RESET_REPOSITORIES:
			return initState
		default:
			return state
	}
}
