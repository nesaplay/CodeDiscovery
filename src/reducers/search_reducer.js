import { 
    SEARCH_SUBMIT, 
    SEARCH_RESET 
} from '../actions/types'

const initState = {
    query: ''
}

export default function(state = initState, action) {
	switch (action.type) {
		case SEARCH_SUBMIT:
			return {
				...state,
				query: action.payload
			}
		case SEARCH_RESET:
			return initState
		default:
			return state
	}
}
