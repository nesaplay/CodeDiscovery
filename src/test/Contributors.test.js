import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'

import Contributors from '../containers/Contributors'
import reducers from '../reducers/index'

const store = createStore(reducers)

it('renders a container with injected store', () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Contributors />
		</Provider>
	)

	expect(wrapper).toMatchSnapshot()
})
