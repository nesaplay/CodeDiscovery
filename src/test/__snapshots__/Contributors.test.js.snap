// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a container with injected store 1`] = `<Connect(Contributors) />`;
